modules/fedimint-ln-common/src/db.rs

```diff
@@ -57,6 +58,41 @@ impl_db_lookup!(
     query_prefix = ContractUpdateKeyPrefix
 );
 
+/// We keep a separate mapping of incoming and outgoing ContractIds to Amounts,
+/// which allows us to quickly audit the total liabilities in the Lightning
+/// module
+#[derive(Debug, Clone, Encodable, Decodable, Serialize, PartialEq)]
+pub enum LightningAuditItemKey {
+    Incoming(ContractId),
+    Outgoing(ContractId),
+}
```

Comment:
The reviewer may notice this differs from `MintAuditItemKey`s, since it doesn't include an aggregate `*Total` key. The motivation for not including the aggregate key is how the Amount associated to the contract [mutates](https://github.com/bradleystachurski/fedimint/blob/4b28e27831ab9c914eb4df68b994e717a5b0d3e1/modules/fedimint-ln-server/src/lib.rs#L624) in the LN module. When a contract reaches a terminal state, the associated amount updates to 0. The additional complexity to update both the individual incoming/outgoing contract audit keys along with the aggregate audit key didn't seem necessary.

In contrast to the mint module, the total number of LN audit keys with a non-zero amount will not grow linearly, so querying the LN audit keys with a non-zero amount should remain quick. I ran benchmark (not included in the PR description) that only uses the audit keys without compaction, which showed the time to run an audit also remained constant. As an aside, that also indicates the root cause for slow audit times is slow de/serialization in the `ContractKey` -> `ContractAccount` db records.

Reply:
Consider putting this comment in the code. :)

Reply:
Consider putting this comment in the code. :)